pandas/tests/series/test_apply.py

```diff
@@ -671,3 +671,16 @@ def test_map_missing_mixed(self, vals, mapping, exp):
         result = s.map(mapping)
 
         tm.assert_series_equal(result, pd.Series(exp))
+
+    @pytest.mark.parametrize("dti,exp", [
+        (Series([1, 2], index=pd.DatetimeIndex([0, 31536000000])),
+            DataFrame(np.repeat([[1, 2]], 2, axis=0), dtype='int64')),
+        (tm.makeTimeSeries(nper=30),
+            DataFrame(np.repeat([[1, 2]], 30, axis=0), dtype='int64'))
```

Comment:
I think there is a 3rd case listed in the OP (of which 2 works fine), can you add that one